coffee-backend/src/main/kotlin/chat/letscoffee/meeting/MeetingService.kt

```diff
@@ -0,0 +1,46 @@
+package chat.letscoffee.meeting
+
+import chat.letscoffee.exception.ResourceNotFoundException
+import chat.letscoffee.exception.ServiceUnavailableException
+import chat.letscoffee.user.User
+import com.github.kittinunf.fuel.Fuel
+import com.github.kittinunf.fuel.gson.responseObject
+import org.springframework.stereotype.Component
+
+
+@Component
+class MeetingService(
+    private val meetingRepository: MeetingRepository
+) {
+
+    fun createMeetingRoom(user: User): Meeting {
+
+        val bodyJson = "{\"Title\" : \"Coffee With Me\"}"
+        val (request, response, result) = Fuel
+            .post("https://api.join.skype.com/v1/meetnow/createjoinlinkguest")
+            .header("Content-Type" to "application/json")
+            .body(bodyJson)
+            .responseObject<MeetingResponse>()
+
+        return result.fold(
+            {
+                var meeting = Meeting(
+                    owner = user,
+                    link = it.joinLink,
+                    provider = MeetingProviderType.SKYPE
+                )
+                meeting = meetingRepository.save(meeting)
+                return@fold meeting
```

Comment:
`return@<method>` scopes the return method to return from the `<method>` instead of the whole function (In this case `fold`). I initially wanted to return the meeting as a normal `return`, but then Kotlin started to complain about not knowing the type of the returned object. The solution was to first return from the `fold` function, then return in the outer scope 🐨